packages/nouns-webapp/src/components/NavWallet/index.tsx

```diff
@@ -0,0 +1,232 @@
+  const { address, buttonStyle } = props;
+
+  const [buttonUp, setButtonUp] = useState(false);
+  const [showConnectModal, setShowConnectModal] = useState(false);
+  const history = useHistory();
+  const { library: provider } = useEthers();
+  const activeAccount = useAppSelector(state => state.account.activeAccount);
+  const { deactivate } = useEthers();
+  const ens = useReverseENSLookUp(address);
+  const shortAddress = useShortAddress(address);
+
+  const showModalHandler = () => {
+    setShowConnectModal(true);
+  };
+  const hideModalHandler = () => {
+    setShowConnectModal(false);
+  };
+
+  const switchWalletHandler = () => {
+    setShowConnectModal(false);
+    setButtonUp(false);
+    deactivate();
+    setShowConnectModal(false);
+    setShowConnectModal(true);
+  };
+
+  const disconectWalletHandler = () => {
+    setShowConnectModal(false);
+    setButtonUp(false);
+    deactivate();
+  };
+
+  const statePrimaryButtonClass = usePickByState(
+    classes.whiteInfo,
+    classes.coolInfo,
+    classes.warnInfo,
+    history,
+  );
+
+  const stateSelectedDropdownClass = usePickByState(
+    classes.whiteInfoSelected,
+    classes.dropdownActive,
+    classes.dropdownActive,
+    history,
+  );
+
+  const mobileTextColor = usePickByState(
+    'rgba(140, 141, 146, 1)',
+    'rgba(121, 128, 156, 1)',
+    'rgba(142, 129, 127, 1)',
+    history,
+  );
+
+  const mobileBorderColor = usePickByState(
+    'rgba(140, 141, 146, .5)',
+    'rgba(121, 128, 156, .5)',
+    'rgba(142, 129, 127, .5)',
+    history,
+  );
+
+  const connectWalletButtonStyle = usePickByState(
+    NavBarButtonStyle.WHITE_WALLET,
+    NavBarButtonStyle.COOL_WALLET,
+    NavBarButtonStyle.WARM_WALLET,
+    history,
+  );
+
+  const customDropdownToggle = React.forwardRef<RefType, Props>(({ onClick, value }, ref) => (
+    <>
+      <div
+        className={`${classes.wrapper} ${
+          buttonUp ? stateSelectedDropdownClass : statePrimaryButtonClass
+        }`}
+        onClick={e => {
+          e.preventDefault();
+          onClick(e);
+        }}
+      >
+        <div className={classes.button}>
+          <div className={classes.icon}>
+            {' '}
+            <Davatar size={21} address={address} provider={provider} />
+          </div>
+          <div>{ens ? ens : shortAddress}</div>
+          <div className={buttonUp ? classes.arrowUp : classes.arrowDown}>
+            <FontAwesomeIcon icon={buttonUp ? faSortUp : faSortDown} />{' '}
+          </div>
+        </div>
+      </div>
+    </>
+  ));
+
+  const CustomMenu = React.forwardRef((props: CustomMenuProps, ref: React.Ref<HTMLDivElement>) => {
+    return (
+      <div
+        ref={ref}
+        style={props.style}
+        className={props.className}
+        aria-labelledby={props.labeledBy}
+      >
+        <div>
+          <div
+            onClick={switchWalletHandler}
+            className={` ${classes.dropDownTop} ${classes.button} ${
```

Comment:
shall we use `clsx` here? check it out https://github.com/nounsDAO/nouns-monorepo/blob/1b9215d196e689115094a6a9c8857db10a115e59/packages/nouns-webapp/src/components/NavBar/index.tsx#L42